### PR TITLE
DOC: add development doc

### LINKED ISSUE #174: DOC: development document

### PR DESCRIPTION
**Pull request by qianduoduo0904**
## Related issue number

Fixes #174 

## Check code requirements

- [ ] tests added / passed (if needed)
- [ ] Ensure all linting tests pass

### CODE REVIEW

**Review by aresnow1 — APPROVED** [review]
LGTM

File: doc/source/getting_started/deployment.rst [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Deployment
+~~~~~~~~~~
+
+Xorbits can be deployed on many platforms, see below for what you need.
+
+========================================= ============================================================
+Deployment                                Description
+========================================= ============================================================
+:ref:`Local <deployment_local>`           Running Xorbits on a local machine, e.g. laptop
```
Comment (UranusSeven) [inline doc/source/getting_started/deployment.rst:9]:
```suggestion
:ref:`Local <deployment_local>`           Run Xorbits on a local machine, e.g.  your laptop
```

File: doc/source/getting_started/deployment.rst [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+:ref:`Cluster <deployment_cluster>`       Deploy Xorbits to existing cluster via command lines
+:ref:`Kubernetes <deployment_kubernetes>` Deploy Xorbits to existing k8s cluster via python codes
```
Comment (UranusSeven) [inline doc/source/getting_started/deployment.rst:11]:
```suggestion
:ref:`Kubernetes <deployment_kubernetes>` Deploy Xorbits to existing k8s cluster via python code
```

File: doc/source/getting_started/deployment.rst [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+:ref:`Cloud <deployment_cloud>`           Deploy Xorbits to various cloud platforms via python codes
```
Comment (UranusSeven) [inline doc/source/getting_started/deployment.rst:12]:
```suggestion
:ref:`Cloud <deployment_cloud>`           Deploy Xorbits to various cloud platforms via python code
```

File: doc/source/getting_started/installation.rst [hunk 4]
```diff
@@ -69,4 +73,18 @@ The following extra dependencies will be installed.
...
+
+Deployment
+~~~~~~~~~~
+
+Xorbits can be deployed on many platforms, see below for what you need.
+
+========================================= ============================================================
+Deployment                                Description
+========================================= ============================================================
+:ref:`Local <deployment_local>`           Running Xorbits on a local machine, e.g. laptop
```
Comment (UranusSeven) [inline doc/source/getting_started/installation.rst:86]:
```suggestion
:ref:`Local <deployment_local>`           Run Xorbits on a local machine, e.g. your laptop
```

File: doc/source/getting_started/installation.rst [hunk 5]
```diff
@@ -69,4 +73,18 @@ The following extra dependencies will be installed.
...
+:ref:`Cluster <deployment_cluster>`       Deploy Xorbits to existing cluster via command lines
+:ref:`Kubernetes <deployment_kubernetes>` Deploy Xorbits to existing k8s cluster via python codes
```
Comment (UranusSeven) [inline doc/source/getting_started/installation.rst:88]:
```suggestion
:ref:`Kubernetes <deployment_kubernetes>` Deploy Xorbits to existing k8s cluster via python code
```

File: doc/source/getting_started/installation.rst [hunk 6]
```diff
@@ -69,4 +73,18 @@ The following extra dependencies will be installed.
...
+:ref:`Cloud <deployment_cloud>`           Deploy Xorbits to various cloud platforms via python codes
```
Comment (UranusSeven) [inline doc/source/getting_started/installation.rst:89]:
```suggestion
:ref:`Cloud <deployment_cloud>`           Deploy Xorbits to various cloud platforms via python codes
```
```suggestion
:ref:`Cloud <deployment_cloud>`           Deploy Xorbits to various cloud platforms via python code
```

File: doc/source/locale/zh_CN/LC_MESSAGES/getting_started/installation.po [hunk 7]
```diff
@@ -27,194 +27,240 @@ msgid ""
...
+
+#: ../../source/getting_started/installation.rst:84
+msgid "Description"
+msgstr ""
+
+#: ../../source/getting_started/installation.rst:86
+msgid ":ref:`Local <deployment_local>`"
+msgstr ""
+
+#: ../../source/getting_started/installation.rst:86
```
Comment (UranusSeven) [inline doc/source/locale/zh_CN/LC_MESSAGES/getting_started/installation.po:239]: need to update .po files

**Review by UranusSeven — APPROVED** [review]
LGTM

**Review by aresnow1 — APPROVED** [review]
LGTM

File: doc/source/locale/zh_CN/LC_MESSAGES/getting_started/installation.po (hunk 7 above)
Comment (qianduoduo0904) [reply doc/source/locale/zh_CN/LC_MESSAGES/getting_started/installation.po:239]: Will regenerate in the next PR.

**Pull request merged**